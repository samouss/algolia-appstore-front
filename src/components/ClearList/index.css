.ClearList__Item {
  justify-content: flex-start;

  & > i {
    margin-right: 10px;
  }
}
