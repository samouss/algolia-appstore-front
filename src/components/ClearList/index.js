import React from 'react';
import PropTypes from 'prop-types';
import FilterList from 'components/FilterList';
import FilterItem from 'components/FilterItem';
import style from './index.css';

const ClearList = ({ refinements, onClear }) => {
  if (!refinements.length) {
    return null;
  }

  return (
    <FilterList>
      <FilterItem
        customClassName={style.ClearList__Item}
        isRefined={false}
        onChange={onClear}
      >
        <i className="fa fa-times-circle" />
        Clear filter(s)
      </FilterItem>
    </FilterList>
  );
};

ClearList.propTypes = {
  refinements: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClear: PropTypes.func.isRequired,
};

export default ClearList;
