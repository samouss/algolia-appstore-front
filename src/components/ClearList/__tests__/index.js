import React from 'react';
import { shallow } from 'enzyme';
import ClearList from '../index';

describe('<ClearList />', () => {
  const defaultProps = {
    refinements: [
      { name: 'category' },
      { name: 'rating' },
    ],
    onClear: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <ClearList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render nothing', () => {
    const props = {
      ...defaultProps,
      refinements: [],
    };

    const component = shallow(
      <ClearList
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onClear', () => {
    const props = {
      ...defaultProps,
      onClear: jest.fn(),
    };

    const component = shallow(
      <ClearList
        {...props}
      />,
    );

    component
      .find('FilterItem')
      .simulate('change');

    expect(props.onClear).toHaveBeenCalled();
  });
});
