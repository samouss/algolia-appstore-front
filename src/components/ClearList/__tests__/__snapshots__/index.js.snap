// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearList /> expect to render 1`] = `
<FilterList
  customClassName=""
>
  <FilterItem
    customClassName="ClearList__Item"
    isRefined={false}
    onChange={[Function]}
  >
    <i
      className="fa fa-times-circle"
    />
    Clear filter(s)
  </FilterItem>
</FilterList>
`;

exports[`<ClearList /> expect to render nothing 1`] = `null`;
